### PR TITLE
Respect Scala 3 distinct module naming

### PR DESCRIPTION
**Pull request by przemek-pokrywka**
Fixes https://github.com/coursier/coursier/issues/1999 (hopefully)

### CODE REVIEW

**Comment by alexarchambault** [comment]
@przemek-pokrywka You need to open a PR with your updated metadata against https://github.com/coursier/test-metadata.

**Comment by przemek-pokrywka** [comment]
Thanks for your guidance @alexarchambault, will try in some free time bit later.

**Comment by alexarchambault** [comment]
@przemek-pokrywka The error seems legit... I can have a look later, but feel free to fix it in the mean time if you'd like.

**Comment by przemek-pokrywka** [comment]
@alexarchambault are you sure that the `coursier.test.CentralTests.forceScalaVersion.force.2.12.8` test case is still up to date? The reason I'm asking is that Coursier 2.0.12 ran manually resolves **the new versions** (not the ones hardcoded in the test expectations).

Here are the failing test logs:
```
Missing element com.lihaoyi:fansi_2.12:0.2.4
Missing element com.lihaoyi:pprint_2.12:0.5.2
Extra element com.lihaoyi:fansi_2.12:0.2.5
Extra element com.lihaoyi:pprint_2.12:0.5.3
Extra element org.scalameta:interactive_2.12.7:4.0.0
Extra element org.scalameta:semanticdb-scalac-core_2.12.7:4.0.0
```
(fansi 0.2.4 ---> 0.2.5, pprint 0.5.2 ---> 0.5.3, ...)

And here's what is the manual resolve returning:
```
$ cs resolve -r jitpack --scala-version 2.12.8 \
> sh.almond:scala-kernel_2.12.8:0.2.2 \
> org.scalameta:interactive_2.12.8:4.1.4 | grep -E \
> 'scalameta:semanticdb-scalac-core|org.scalameta:interactive|pprint|fansi'
com.lihaoyi:fansi_2.12:0.2.5:default
com.lihaoyi:pprint_2.12:0.5.3:default
org.scalameta:interactive_2.12.7:4.0.0:default
org.scalameta:interactive_2.12.8:4.1.4:default
org.scalameta:semanticdb-scalac-core_2.12.7:4.0.0:default
org.scalameta:semanticdb-scalac-core_2.12.8:4.1.4:default
```

**Comment by przemek-pokrywka** [comment]
@alexarchambault adding to the [previous comment](#issuecomment-798975835), the extra dependencies seem to be dragged in by:
```
├─ sh.almond:scala-kernel_2.12.8:0.2.2
[...]
│  ├─ sh.almond:kernel_2.12:0.2.2
[...]
│  │  └─ sh.almond:interpreter_2.12:0.2.2
[...]
│  │     ├─ org.scalameta:metabrowse-server_2.12:0.2.1
[...]
│  │     │  ├─ org.scalameta:interactive_2.12.7:4.0.0
│  │     │  │  ├─ com.lihaoyi:pprint_2.12:0.5.3
│  │     │  │  │  ├─ com.lihaoyi:fansi_2.12:0.2.5
│  │     │  │  │  │  ├─ com.lihaoyi:sourcecode_2.12:0.1.4 -> 0.1.5
│  │     │  │  │  │  │  └─ org.scala-lang:scala-library:2.12.6 -> 2.12.8
│  │     │  │  │  │  └─ org.scala-lang:scala-library:2.12.0 -> 2.12.8
│  │     │  │  │  ├─ com.lihaoyi:sourcecode_2.12:0.1.4 -> 0.1.5
│  │     │  │  │  │  └─ org.scala-lang:scala-library:2.12.6 -> 2.12.8
│  │     │  │  │  └─ org.scala-lang:scala-library:2.12.2 -> 2.12.8
```
(obtained with `cs resolve --tree -r jitpack --scala-version 2.12.8 sh.almond:scala-kernel_2.12.8:0.2.2 org.scalameta:interactive_2.12.8:4.1.4`)

**Comment by przemek-pokrywka** [comment]
@alexarchambault I think you were correct. CentralTest passes after the newest fix.

File: modules/core/shared/src/main/scala/coursier/core/Resolution.scala [hunk 1]
```diff
@@ -592,11 +592,12 @@ object Resolution {
   def forceScalaVersion(sv: String): Dependency => Dependency = {
 
     val sbv = sv.split('.').take(2).mkString(".")
+    val scala = if (sbv.startsWith("3")) "scala3" else "scala"
 
     val scalaModules = Set(
-      ModuleName("scala-library"),
```
Comment (alexarchambault) [inline modules/core/shared/src/main/scala/coursier/core/Resolution.scala:597]: Maybe we could not force a version for `scala-reflect` and `scalap`, when Scala 3 is used, like
```scala
val scalaModules =
  if (sbv.startsWith("3."))
    Set(
      "scala3-library",
      "scala3-compiler"
    )
  else
    Set(
      "scala-library",
      "scala-compiler",
      "scala-reflect",
      "scalap"
    )
```

**Review by alexarchambault — COMMENTED** [review]
Thanks @przemek-pokrywka! I only one minor comment.

About the failing job, https://github.com/coursier/coursier/pull/2006 ought to fix that.

File: modules/coursier/shared/src/main/scala/coursier/Resolve.scala [hunk 2]
```diff
@@ -215,12 +215,18 @@ object Resolve extends PlatformResolve {
...
-        Seq(
-          Module(scalaOrg, ModuleName("scala-library")) -> params.selectedScalaVersion,
-          Module(scalaOrg, ModuleName("org.scala-lang:scala-reflect")) -> params.selectedScalaVersion,
-          Module(scalaOrg, ModuleName("org.scala-lang:scala-compiler")) -> params.selectedScalaVersion,
-          Module(scalaOrg, ModuleName("org.scala-lang:scalap")) -> params.selectedScalaVersion
-        )
+        if (params.selectedScalaVersion.startsWith("3"))
+          Seq(
+            Module(scalaOrg, ModuleName("scala3-library")) -> params.selectedScalaVersion,
+            Module(scalaOrg, ModuleName("org.scala-lang:scala3-compiler")) -> params.selectedScalaVersion
```
Comment (alexarchambault) [inline modules/coursier/shared/src/main/scala/coursier/Resolve.scala:221]: I think the former code was wrong 😅, the organization shouldn't be here:
```suggestion
            Module(scalaOrg, ModuleName("scala3-compiler")) -> params.selectedScalaVersion
```

File: modules/coursier/shared/src/main/scala/coursier/Resolve.scala [hunk 3]
```diff
@@ -215,12 +215,18 @@ object Resolve extends PlatformResolve {
...
+          )
+        else
+          Seq(
+            Module(scalaOrg, ModuleName("scala-library")) -> params.selectedScalaVersion,
+            Module(scalaOrg, ModuleName("org.scala-lang:scala-compiler")) -> params.selectedScalaVersion,
+            Module(scalaOrg, ModuleName("org.scala-lang:scala-reflect")) -> params.selectedScalaVersion,
+            Module(scalaOrg, ModuleName("org.scala-lang:scalap")) -> params.selectedScalaVersion
```
Comment (alexarchambault) [inline modules/coursier/shared/src/main/scala/coursier/Resolve.scala:228]: Same as above
```suggestion
            Module(scalaOrg, ModuleName("scala-compiler")) -> params.selectedScalaVersion,
            Module(scalaOrg, ModuleName("scala-reflect")) -> params.selectedScalaVersion,
            Module(scalaOrg, ModuleName("scalap")) -> params.selectedScalaVersion
```

**Review by alexarchambault — COMMENTED** [review]
It seems some of the former code was wrong. I hope the suggested changes in this review don't break anything 😬

File: modules/coursier/shared/src/main/scala/coursier/Resolve.scala (hunk 2 above)
Comment (przemek-pokrywka) [reply modules/coursier/shared/src/main/scala/coursier/Resolve.scala:221]: Glad you've noticed that; I was a bit wondered but thought that it's a part of some plan :smile:

**Comment by przemek-pokrywka** [comment]
By the way, is it justified to keep two similar checks (if Scala 3 then ...) in two separate places?

**Comment by alexarchambault** [comment]
> By the way, is it justified to keep two similar checks (if Scala 3 then ...) in two separate places?

Not particularly... feel free to factor it if you'd like.

**Comment by przemek-pokrywka** [comment]
I would leave it as it is right now, thanks. I mostly asked for my understanding; I was curious if there was any plan behind that.

I'll appreciate it if the changes can be released. I'd be able to use them in TSK, for Scala 3 examples :smile:

**Review by alexarchambault — APPROVED** [review]
Merging, thanks @przemek-pokrywka!

**Pull request merged**